lnrpc/routerrpc/router_server.go

```diff
@@ -709,8 +709,22 @@ func (s *Server) QueryProbability(ctx context.Context,
 
 	amt := lnwire.MilliSatoshi(req.AmtMsat)
 
+	// Compute the probability.
+	var prob float64
 	mc := s.cfg.RouterBackend.MissionControl
-	prob := mc.GetProbability(fromNode, toNode, amt)
+	capacity, err := s.cfg.RouterBackend.FetchAmountPairCapacity(
+		fromNode, toNode, amt,
+	)
+
+	// If we cannot query the capacity this means that either we don't have
+	// information available or that the channel fails min/maxHtlc
+	// constraints, so we return a zero probability.
+	if err != nil {
+		log.Errorf("Cannot fetch capacity: %v", err)
+	} else {
+		prob = mc.GetProbability(fromNode, toNode, amt, capacity)
+	}
+
```

Comment:
More generally I am wondering what this `QueryProbability` is actually used for in the real world and whether it is worth the maintenance that it requires. This is just a side comment.

Reply:
What would you think of just marking it as deprecated in 0.16, and then remove the rpc plus `FetchAmountPairCapacity` in 0.17?

Perhaps `QueryMissionControlRequest` can be extended with from/to fields to return the base data for the probability calculation. If someone really needs a probability, they can always link against lnd "the library" and calculate it client-side.

Reply:
> I am wondering what this QueryProbability is actually used for in the real world

Wondering about this too, it looks like right now it's only used for debugging. I added a commit to mark it deprecated in the rpc and lncli, hope I found all the places.